zeek/plugin/src/runtime-support.cc

```diff
@@ -86,9 +86,15 @@ void rt::raise_event(EventHandlerPtr handler, const hilti::rt::Vector<Val*>& arg
     ::mgr.QueueEventFast(handler, vl);
 }
 
-BroType* rt::event_arg_type(EventHandlerPtr handler, uint64_t idx) {
+BroType* rt::event_arg_type(EventHandlerPtr handler, uint64_t idx, std::string_view location) {
     assert(handler);
-    return (*handler->FType()->ArgTypes()->Types())[idx];
+
+    auto zeek_args = handler->FType()->ArgTypes()->Types();
+    if ( idx >= zeek_args->length() )
+        throw TypeMismatch(fmt("more parameters given than the %u that the Zeek event expects", zeek_args->length()),
```

Comment:
Just wondering, since we now abort processing for an incorrect number of arguments, can we remove turn the check in `rt::raise_event` into an assert/hard failure, or remove it altogether?

Reply:
No, there are still cases where get in there (less parameters on the Spicy side than Zeek)